### ISSUE TITLE
Unable to Scrape Angellist 

### ISSUE THREAD

**Issue by cancan101** [Anorov/cloudflare-scrape#273]
Before creating an issue, first upgrade cfscrape with `pip install -U cfscrape` and see if you're still experiencing the problem.

## Version number

Run `pip show cfscrape` and paste the output below:

```
Name: cfscrape
Version: 2.0.7
Summary: A simple Python module to bypass Cloudflare's anti-bot page. See https://github.com/Anorov/cloudflare-scrape for more information.
Home-page: https://github.com/Anorov/cloudflare-scrape
Author: Anorov
Author-email: anorov.vorona@gmail.com
License: UNKNOWN
Location: /Users/alex/.pyenv/versions/3.7.3/envs/xxxx/lib/python3.7/site-packages
Requires: requests
Required-by:
```

## Code snippet experiencing the issue

```
import cfscrape
scraper = cfscrape.create_scraper()
scraper.get("https://angel.co/company/clutter/jobs/")
```

## Complete exception and traceback

(*If the problem doesn't involve an exception being raised, leave this blank*)

```
/lib/python3.7/site-packages/cfscrape/__init__.py in handle_captcha_challenge(self, resp, url)
    145             error += ". Your OpenSSL version is lower than 1.1.1. Please upgrade your OpenSSL library and recompile Python."
    146 
--> 147         raise CloudflareCaptchaError(error, response=resp)
    148 
    149     def solve_cf_challenge(self, resp, **original_kwargs):

CloudflareCaptchaError: Cloudflare captcha challenge presented for angel.co (cfscrape cannot solve captchas)
```

## URL of the Cloudflare-protected page

`https://angel.co/company/clutter/jobs/`

## URL of Pastebin/Gist with HTML source of protected page

https://www.npmjs.com/package/cloudscraper is able to scrape the page

**Comment by nm111** [comment]
Please read the error message. cfscrape cannot solve captchas.

**Comment by cancan101** [comment]
Right but the linked  js library is able to access the page without triggering the captcha defense.

**Comment by pro-src** [comment]
Hi @cancan101,

This is more than likely TLS related. Unlike python, Node.js links to OpenSSL statically and doesn't use your system's version of openssl by default.

`node -p process.versions.openssl`
`python -c 'import ssl; print(ssl.OPENSSL_VERSION)'`

If you can upgrade openssl and that fixes the problem, great, otherwise see #250, #264, and similar issues.
You might have a look at https://github.com/codemanki/cloudscraper/issues/234 as well.

I'm sure one of those issues will detail a suitable workaround.

**Comment by nm111** [comment]
I can reproduce the issue with upgraded openssl
```
$ node -p process.versions.openssl
1.1.1b
$ python -c 'import ssl; print(ssl.OPENSSL_VERSION)'
OpenSSL 1.1.1b  26 Feb 2019
```

**Comment by pro-src** [comment]
I can't reproduce the issue with upgraded openssl. I don't get a CAPTCHA at all but that's not unusal...
```
root@kali:~# node -p process.versions.openssl
1.1.1c
root@kali:~# python -c 'import ssl; print(ssl.OPENSSL_VERSION)'
OpenSSL 1.1.1c  28 May 2019
```
Please see the linked issues.

**Comment by cancan101** [comment]
I have a newer version of OpenSSL on Python as I had rebuilt the installed version:
```
$ node -p process.versions.openssl
1.1.0j
$ python -c 'import ssl; print(ssl.OPENSSL_VERSION)'
OpenSSL 1.1.1c  28 May 2019
```

**Comment by pro-src** [comment]
There seems to be a miscommunication.

> If you can upgrade openssl and that fixes the problem, great, otherwise...

Please notice the conjunction above. I have not stated that a particular OpenSSL version is the cause of this problem. Why are we talking about OpenSSL versions then? Users have reported that changing their version of OpenSSL resolved the issue for them. It should be updated regardless.

If the information that you're looking for is not in the linked issues, I'll be more than happy to try and answer your question(s).

**Comment by pro-src** [comment]
Have a look at #268 and let me know if you read any of the issues, what you've tried, etc. 

> If you can upgrade openssl and that fixes the problem, great, otherwise...

Please notice the conjunction above. I have not stated that a particular OpenSSL version is the cause of this problem. Why are we talking about OpenSSL versions then? Users have reported that changing their version of OpenSSL resolved the issue for them. It should be updated regardless.

If the information that you're looking for is not in the linked issues, I'll be more than happy to try and answer your question(s).

**Comment by pro-src** [comment]
cfscrape, cloudscraper, etc. all impersonate the chromium browser by using it's UA info, down to the order that HTTP headers are sent. Cloudflare expects the TLS to match that of Chromium and is checking for outdated things.

Problem:
1. Chromium uses boringssl and tweaks it's signature algorithms (https://bugs.chromium.org/p/chromium/issues/detail?id=655318)
2. Neither Python, Node,  or Go expose a way to set those signature algorithms.
3. Everybody is trying to remove ciphers from the default cipher suite to change what would be in the signature algorithms as a side effect.
4. The signature algorithms can be different with the same version of openssl, and with the same default cipher suite. OpenSSL may behave differently depending on CPU features.

Solution(s):
Node could be compiled with boringssl. Electron does this but they're also patching things...
Python could use pyCURL compiled with boringssl (shouldn't be too hard) but you'll have to modify cfscrape to use pyCURL instead of requests.
Expose `SSL_CTX_set1_client_sigalgs` and recompile python or Node.js (Which ever you're using).

Workaround(s):
Use a different cipher list and hope that the side effects will result in certain signature algorithms being omitted. (See the linked and similar issues)

**Comment by pro-src** [comment]
@cancan101,

I would try using 1.1.0j but the workarounds are much easier. If you want to go that route anyway, grab a tarball of 1.1.0j and see https://superuser.com/questions/1346141/how-to-link-python-to-the-manually-compiled-openssl-rather-than-the-systems-one

I mentioned not needing to recompile python in another issue, mostly wrong. Anyways...
